src/SmartTransactionsController.test.ts

```diff
@@ -1,44 +1,74 @@
 import nock from 'nock';
 import { NetworkState } from '@metamask/network-controller';
+import { convertHexToDecimal } from '@metamask/controller-utils';
 import SmartTransactionsController, {
   DEFAULT_INTERVAL,
 } from './SmartTransactionsController';
 import { API_BASE_URL, CHAIN_IDS } from './constants';
 import { SmartTransaction, SmartTransactionStatuses } from './types';
-
-const confirmExternalMock = jest.fn();
+import * as utils from './utils';
+
+/**
+ * Resolve all pending promises.
+ * This method is used for async tests that use fake timers.
+ * See https://stackoverflow.com/a/58716087 and https://jestjs.io/docs/timer-mocks.
+ */
+function flushPromises(): Promise<unknown> {
+  return new Promise(jest.requireActual('timers').setImmediate);
+}
+// const confirmExternalMock = jest.fn();
 
 jest.mock('@ethersproject/bytes', () => ({
   ...jest.requireActual('@ethersproject/bytes'),
   hexlify: (str: string) => `0x${str}`,
 }));
 
-jest.mock('@ethersproject/providers', () => ({
-  Web3Provider: class Web3Provider {
-    getBalance = () => ({ toHexString: () => '0x1000' });
-
-    getTransactionReceipt = jest.fn(() => ({ blockNumber: '123' }));
-
-    getTransaction = jest.fn(() => ({
-      maxFeePerGas: { toHexString: () => '0x123' },
-      maxPriorityFeePerGas: { toHexString: () => '0x123' },
-    }));
-
-    getBlock = jest.fn();
-  },
-}));
+jest.mock('@metamask/eth-query', () => {
+  return class FakeEthQuery {
+    sendAsync = jest.fn(({ method }, callback) => {
+      switch (method) {
+        case 'getBalance': {
```

Comment:
Should we prepend these methods with `eth_`? Also, I believe `getTransaction` should be `eth_getTransactionByHash` (see comment I left below).

Reply:
Looks like there are a bunch of instances in various controllers where the methods are not [not prepended with `eth_`](https://github.com/MetaMask/core/blob/d992b424f934c4a96983c9a5005a44958e02979f/packages/assets-controllers/src/AccountTrackerController.ts#L197). Other examples [here](https://github.com/MetaMask/core/blob/d992b424f934c4a96983c9a5005a44958e02979f/packages/gas-fee-controller/src/gas-util.ts#L123), [here](https://github.com/MetaMask/core/blob/d992b424f934c4a96983c9a5005a44958e02979f/packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory/fetchLatestBlock.ts#L18) & [here](https://github.com/MetaMask/core/blob/d992b424f934c4a96983c9a5005a44958e02979f/packages/transaction-controller/src/utils/gas.ts#L69)

Reply:
Also seems like you could just call the methods right off the ethQuery class if you wanted based [on the readme](https://github.com/MetaMask/eth-query#readme)

Reply:
Ah, different definitions of "method" :)

The _instance_ methods on EthQuery aren't prepended with `eth_`, but when the request is made through `sendAsync`, the `eth_` prefix is added to the _RPC_ method. So if we are mocking `sendAsync` on EthQuery, it seems like we ought to use the correct RPC method. Otherwise, I'm not sure how these tests are passing 🤔

Reply:
Okay, I dug into this a bit. When using `query` in the implementation file, we're passing the name of an EthQuery instance method, and so `query` will then try to call a method on the EthQuery instance that matches that name. But because the only method we have in our fake EthQuery class is `sendAsync`, `query` will then call `sendAsync` and just pass along the method it has: https://github.com/MetaMask/core/blob/306eecda2ef1d0ebcf4df3b8e56338c0f16ceac9/packages/controller-utils/src/util.ts#L432-L454. Hence, `method` here _doesn't_ have an `eth_` prefix.

This accidentally works because none of the strings in the `switch` statement have that prefix. But I think it might be confusing if people are familiar with how EthQuery is supposed to work.

Therefore would it make more sense to bring in `EthQuery` via `jest.requireActual`, `extend` from it, and then update all of these RPC methods to have an `eth_` prefix? (I can make a new suggestion)